examples/collector/README.md

```diff
@@ -0,0 +1,114 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+## Snap Plugin Library for C++: Collector Plugin Example
+Here you will find an example plugin that covers the basics for writing a collector plugin.
+
+## Plugin Naming, Files, and Directory
+For your collector plugin, create a new repository and name your plugin project using the following format:
+
+>snap-plugin-[plugin-type]-[plugin-name]
+
+For example:
+>snap-plugin-collector-rando
+
+Proposed files and directory structure:  
+```
+snap-plugin-[plugin-type]-[plugin-name]
+ |--src
+  |--[plugin-name].cc  
+  |--[plugin-name]_test.cc  
+```
+
+For example:
+```
+snap-plugin-collector-rando
+ |--src
+  |--rando.cc  
+  |--rando_test.cc  
+```
+
+## Interface methods
+
+In order to write a plugin for Snap, it is necessary to define a few methods to satisfy the appropriate interfaces. These interfaces must be defined for a collector plugin:
+
+```cpp
+/**
+ * The interface for a collector plugin.
+ * A Collector is the source.
+ * It is responsible for collecting metrics in the snap pipeline.
+ */
+class CollectorInterface : public PluginInterface {
+public:
+  Type GetType() const final;
+  CollectorInterface* IsCollector() final;
+  
+  /*
+   * (inherited from PluginInterface)
+   */
+  virtual const ConfigPolicy get_config_policy() = 0;
+
+  /*
+   * get_metric_types should report all the metrics this plugin can collect.
+   */
+  virtual std::vector<Metric> get_metric_types(Config cfg) = 0;
+
+  /*
+   * collect_metrics is given a list of metrics to collect.
+   * It should collect and annotate each metric with the apropos context.
+   */
+  virtual void collect_metrics(std::vector<Metric> &metrics) = 0;
+
+};
+```
+
+The interface is slightly different depending on what type (collector, processor, or publisher) of plugin is being written. Please see other plugin types for more details.
+
+## Starting a plugin
+
+After implementing a type that satisfies one of {collector, processor, publisher} interfaces, all that is left to do is to call the appropriate plugin.start_xxx() with your plugin specific meta options. For example with no meta options specified:
+
+```cpp
+    Plugin::start_collector(&plg, Meta{Type::Collector, "rando", 1});
+```
+
+### Meta options
+
+The available options are defined in [src/snap/plugin.h](https://github.com/intelsdi-x/snap-plugin-lib-cpp/tree/master/src/snap/plugin.h). This structure defines default values.
+Plugin developer needs to define only plugin Type, name and version. 
+
+An example using some arbitrary values::
+
+```cpp
+    Plugin::start_collector(&plg, Meta{Type::Collector, "rando", 1});
+    Rando plg = Rando();
+    Meta meta = Meta{Type::Collector, "rando", 1};
+    meta.exclusive = true;
+    Plugin::start_collector(&plg, meta);
+```
+
+## Testing
+
+Official Snap plugins differentiate tests by scope into "small", "medium" and "large".
```

Comment:
Would you like to put here referenece to Snap's [testing quidelines](https://github.com/intelsdi-x/snap/blob/master/CONTRIBUTING.md#testing-guidelines)